contracts/protocol/libraries/logic/SupplyLogic.sol

```diff
@@ -173,6 +174,19 @@ library SupplyLogic {
             DataTypes.AssetType.ERC721
         );
 
+        INToken nToken = INToken(reserveCache.xTokenAddress);
+        if (nToken.getXTokenType() == XTokenType.NTokenUniswapV3) {
```

Comment:
this kind of checks doesn't seem like a scalable solution. I do prefer having a separate supply function for uniswapV3 comparing to this ad-hoc solution.

Reply:
if so, then we have too much repeated code logic for supply, withdraw, setUserAsCollateral, finalizeTransfer and liquidate.
Most important, we don't have a strong scalability requirement right now, and if we do have in the future, we can change at that moment.

Reply:
Even if univ3's interface is separated, we need additional checks to prevent users supplying univ3 NFTS using this interface